ranger-discovery-bundle/src/main/java/io/appform/ranger/discovery/bundle/id/nonce/RandomNonceGenerator.java

```diff
@@ -0,0 +1,139 @@
+package io.appform.ranger.discovery.bundle.id.nonce;
+
+import com.google.common.base.Strings;
+import dev.failsafe.Failsafe;
+import dev.failsafe.FailsafeExecutor;
+import dev.failsafe.RetryPolicy;
+import io.appform.ranger.discovery.bundle.id.CollisionChecker;
+import io.appform.ranger.discovery.bundle.id.Constants;
+import io.appform.ranger.discovery.bundle.id.Domain;
+import io.appform.ranger.discovery.bundle.id.GenerationResult;
+import io.appform.ranger.discovery.bundle.id.IdInfo;
+import io.appform.ranger.discovery.bundle.id.IdValidationState;
+import io.appform.ranger.discovery.bundle.id.constraints.IdValidationConstraint;
+import io.appform.ranger.discovery.bundle.id.formatter.IdFormatter;
+import io.appform.ranger.discovery.bundle.id.request.IdGenerationRequest;
+import lombok.val;
+import org.joda.time.DateTime;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+
+public class RandomNonceGenerator extends NonceGeneratorBase {
+    private final FailsafeExecutor<GenerationResult> RETRYER;
+
+    public RandomNonceGenerator(final IdFormatter idFormatter) {
+        super(idFormatter);
+        RetryPolicy<GenerationResult> RETRY_POLICY = RetryPolicy.<GenerationResult>builder()
+                .withMaxAttempts(readRetryCount())
+                .handleIf(throwable -> true)
+                .handleResultIf(Objects::isNull)
+                .handleResultIf(generationResult -> generationResult.getState() == IdValidationState.INVALID_RETRYABLE)
+                .onRetry(event -> {
```

Comment:
This is supposed to be a listener. Performing mutations in this is a very bad idea

Reply:
Also i think this is being done twice. Once here once in get() method